src/main/java/com/liferay/faces/util/product/internal/ProductCDIApiImpl.java

```diff
@@ -45,6 +45,8 @@ public ProductCDIApiImpl() {
 			else {
 				logger.warn("Unable to obtain version information for {0}.", this.title);
 			}
+
+			initStringValue(version);
```

Comment:
@stiemannkj1 The pattern of calling `initStringValue(version)` happens quite frequently throughout many of these classes. It looks a little non-obvious to pass the `protected String version` that is defined in ProductBaseImpl. Here's one idea: Perhaps an abstract method called `postConstruct()` could be defined in ProductBaseImpl:
```
protected void postConstruct() {
    initStringValue(version);
}
```
And then each constructor can call that as the very last thing? Also, the `postConstruct()` method can be overridden in subclasses if necessary.